API.md

```diff
@@ -1,64 +1,283 @@
+
+```javascript
+frisby.create('a test')
+    .get('http://example.com')
+    .expectStatus(200)
+    // any number of additional expect statements here
+    .toss();
+```
+
+## Commands
+
+These are the commands you'll need to get IcedFrisby talking HTTP.
+
+An optional parameters object is accepted by each of these methods. All object parameters are optional.
+
+```javascript
+{
+  json: 'boolean',         //Whether this will be a JSON body. Overrides value set in globalSetup().
+  body: 'string'|'object', //The body to include in the outbound request. This overrides "data" if provided in method call.
+  mock: 'function',        //A mock runner to use. When not provided, uses "request" (i.e. does it for real).
+  form: 'boolean'          //Use the object in the body to create a form-encoded request.
+}
+```
+
+### get(uri, params)
+
+| Parameter | Description | Required |
+| --------- | ----------- | -------- |
+| uri | The URI/URL being requested | Yes |
+| params | An optional params object, as described above | No |
+
+Perform an HTTP GET on the specified URI.
+
+```javascript
+frisby.create('a test')
+    .get('http://example.com/login')
+    .expectStatus(200)
+    // any number of additional expect statements here
+    .toss();
+```
+
+### head(uri, params)
+
+Identical to [get](#geturi-params), using an HTTP HEAD request.
+
+### options(uri, params)
+
+Identical to [get](#geturi-params), using an HTTP OPTIONS request.
+
+### post(uri, data, params)
+
+| Parameter | Description | Required |
+| --------- | ----------- | -------- |
+| uri | The URI string | Yes |
+| data | The data to post. An object when `{ json: true }` or `{ form: true }`, or else a data string / buffer. | Yes |
+| params | An optional params object, as described above. | No |
+
+Perform an HTTP POST to the specified URI.
+
+```javascript
+frisby.create('a test')
+    .post('http://example.com/account', {
+      username: 'joe@example.com',
+      password: 'J0£_£x@mpl£'
+    })
+    .expectStatus(200)
+    // any number of additional expect statements here
+    .toss();
+
+frisby.create('another test')
+    .post('http://example.com/contact-us', {
+      name: 'Test Person',
+      comment: 'What a lovely test'
+    }, {form: true})
+    .expectStatus(200)
+    // any number of additional expect statements here
+    .toss();
+
+```
+
+### put(uri, data, params)
+
+Identical to [post](#posturi-params), using an HTTP PUT request.
+
+### patch(uri, data, params)
+
+Identical to [post](#posturi-params), using an HTTP PATCH request.
+
+### delete(uri, data, params)
+
+Identical to [post](#posturi-params), using an HTTP DELETE request.
+
+### addHeader(header, content)
+
+Adds an HTTP header to your request
+
+| Parameter | Description | Required |
+| --------- | ----------- | -------- |
+| header | String giving the name of the header. Case insensitive. | Yes |
+| content | String giving the value of the given header. Case sensitive. | Yes |
```

Comment:
Note that you can invoke this multiple times per header?

Reply:
Done.